numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -3124,11 +3132,16 @@ VOID_compare(char *ip1, char *ip2, PyArrayObject *ap)
                     new->f->copyswap(nip1, NULL, swap, dummy);
             }
             if (swap || !npy_is_aligned(nip2, new->alignment)) {
-                /* create buffer and copy */
-                nip2 = npy_alloc_cache(new->elsize);
+                /*
+                 * create temporary buffer and copy,
+                 * always use the current handler for internal allocations
+                 */
+                nip2 = PyDataMem_UserNEW(new->elsize, &mem_handler->allocator);
                 if (nip2 == NULL) {
                     if (nip1 != ip1 + offset) {
-                        npy_free_cache(nip1, new->elsize);
+                        /* destroy temporary buffer */
+                        PyDataMem_UserFREE(nip1, new->elsize,
+                                           &mem_handler->allocator);
```

Comment:
I didn't want to do this dance:
```
mem_handler = PyDataMem_GetHandler(ap);
if (mem_handler == NULL) {
    mem_handler = PyDataMem_GetHandler(NULL);
    if (mem_handler == NULL) {
        <handle error>
    }
}
```

Do you think it is worthwhile? This pattern occurs in `_new_sortlike()` and `_newargsortlike` (in `numpy/core/src/multiarray/item_selection.c`) as well.

Reply:
I don't know.  I started using the Python allocators for temporary allocations (since ABI doesn't matter), but with this work there may also be some use in honoring the current allocator (i.e. it will end up on the same memory region?).  In any case, I don't think we have to worry about it right now.